### PR TITLE
Prepare Elasticsearch 2.x upgrade

### PR DESCRIPTION
**Pull request by diox**
This is the first pass towards making our code compatible with ES 1.6 and 2.x at the same time. Some work is still needed (like replacing facets with aggregation) but this will be done in a later PR.

Supports #2850 (replaces #2883).

### CODE REVIEW

File: src/olympia/addons/indexers.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def get_mapping(cls):
                 'properties': {
                     'id': {'type': 'long'},
 
-                    'app': {'type': 'long'},
+                    'app': {'type': 'byte'},
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: Why is this changing if I mind asking? afair es 2.x still supports `long`?

File: src/olympia/lib/es/management/commands/reindex.py [hunk 2]
```diff
@@ -178,12 +179,17 @@ def add_alias_action(action, index, alias):
...
-                add_alias_action('remove', old_index, alias)
+            try:
+                olds = ES.indices.get_aliases(alias)
+                for old_index in olds:
+                    # Mark the index to be removed later.
+                    to_remove.append(old_index)
+                    # Mark the alias to be removed from that index.
+                    add_alias_action('remove', old_index, alias)
+            except NotFoundError:
+                # If the alias dit not exist, ignore it, don't try to remove
```
Comment (EnTeQuAk) [inline src/olympia/lib/es/management/commands/reindex.py]: We should at least log the 404 imho to make sure we're able to see why things are happening or why they aren't.

**Comment by EnTeQuAk** [comment]
Looks great so far to me, only some minor questions. r+wc

File: src/olympia/addons/indexers.py (hunk 1 above)
Comment (diox) [inline src/olympia/addons/indexers.py]: The same field is a `byte` elsewhere, and in 2.x field definitions need to be consistent across different document types on the same index (yes, it's annoying). On the plus side, `app` should really be a `byte`, there is no reason to use a `long`.

**Pull request merged**